src/kOS.Safe/Screen/ScreenBuffer.cs

```diff
@@ -38,6 +44,12 @@ public class ScreenBuffer : IScreenBuffer
         public int RowCount { get; private set; }
         public int ColumnCount { get; private set; }
 
+        /// <summary>Delegate prototype expected by AddResizeNotifier</summary>
+        /// <param name="sb">This screenbuffer telling the callback who it is</param>
+        /// <returns>telling this screenbuffer how many vertical rows to scroll as a result of the resize.</returns>
+        public delegate int ResizeNotifier(IScreenBuffer sb);
```

Comment:
This integer return value is just hardcoded to return zero in all the delegates that use the system at the moment.
The intent is to allow a delegate to inform the parent ScreenBuffer that it needs to scroll itself based upon the resize operation, to keep the important part in view (the part with the cursor).  Right now that calculation wasn't quite right and it's not strictly necessary for a first pass at this so I left them hardcoded to 0.  (The effect is that at the moment, if you shrink the width such that the current command line gets pushed off the bottom of the screen, you don't scroll the cursor back into view until you start typing, which snaps it back into view.)